src/main/kotlin/racing/domain/RaceProcessor.kt

```diff
@@ -0,0 +1,32 @@
+package racing.domain
+
+class RaceProcessor {
+    fun execute(
+        cars: List<Car>,
+        round: Int,
+    ): RaceResult {
+        validate(cars, round)
+
+        return start(cars, round)
+    }
+
+    private fun start(
+        cars: List<Car>,
+        round: Int,
+    ): RaceResult {
+        return List(round) {
+            cars.forEach(Car::move)
+            Result.of(it + 1, cars)
```

Comment:
```suggestion
            Result.of(it++, cars)
```

요건 표현하는 방식이 3개가 있을 듯해요~

- it++
- it + 상수화 된 +1
- 주석으로 설명 표기

Reply:
스코프에서 사용되는 it 는 value라서 후치 연산자는 안되는 것 같아요
해당 객체 내의 상수로 고민해볼게요